release/cli/pkg/operations/download.go

```diff
@@ -44,139 +47,162 @@ func DownloadArtifacts(r *releasetypes.ReleaseConfig, eksArtifacts map[string][]
+		if err != nil {
+			return errors.Cause(err)
+		}
+
+		// Download checksum files for the archive
+		checksumExtensions := []string{
+			".sha256",
+			".sha512",
+		}
+
+		// Adding a special case for tinkerbell/hook project.
+		// The project builds linux kernel files that are not stored as tarballs and currently do not have SHA checksums.
+		// TODO(pokearu): Add logic to generate SHA for hook project
+		if artifact.Archive.ProjectPath == constants.HookProjectPath {
+			checksumExtensions = []string{}
+		}
+
+		for _, extension := range checksumExtensions {
+			objectShasumFileName := fmt.Sprintf("%s%s", sourceS3Key, extension)
+			objectShasumFileKey := filepath.Join(sourceS3Prefix, objectShasumFileName)
+			objectShasumFileLocalFilePath := filepath.Join(artifactPath, objectShasumFileName)
+			fmt.Printf("Checksum file - %s\n", objectShasumFileKey)
+
+			err := s3Retrier.Retry(func() error {
+				if !s3.KeyExists(r.SourceBucket, objectShasumFileKey) {
+					return fmt.Errorf("requested object not found")
+				}
+				return nil
+			})
+			if err != nil {
+				if r.BuildRepoBranchName != "main" {
+					var latestSourceS3PrefixFromMain string
+					fmt.Printf("Artifact corresponding to %s branch not found for %s archive. Using artifact from main\n", r.BuildRepoBranchName, sourceS3Key)
+					if strings.Contains(sourceS3Key, "eksctl-anywhere") {
+						latestSourceS3PrefixFromMain = strings.NewReplacer(r.CliRepoBranchName, "latest").Replace(sourceS3Prefix)
+					} else {
+						gitTagFromMain, err := filereader.ReadGitTag(artifact.Archive.ProjectPath, r.BuildRepoSource, "main")
 						if err != nil {
 							return errors.Cause(err)
 						}
-						latestSourceS3PrefixFromMain := strings.NewReplacer(r.BuildRepoBranchName, "latest", artifact.Manifest.GitTag, gitTagFromMain).Replace(sourceS3Prefix)
-						objectKey = filepath.Join(latestSourceS3PrefixFromMain, sourceS3Key)
-					} else {
-						return fmt.Errorf("retries exhausted waiting for archive to be uploaded to source location: %v", err)
+						latestSourceS3PrefixFromMain = strings.NewReplacer(r.BuildRepoBranchName, "latest", artifact.Archive.GitTag, gitTagFromMain).Replace(sourceS3Prefix)
 					}
+					objectShasumFileKey = filepath.Join(latestSourceS3PrefixFromMain, objectShasumFileName)
+				} else {
+					return fmt.Errorf("retries exhausted waiting for checksum file to be uploaded to source location: %v", err)
 				}
+			}
 
-				err = s3.DownloadFile(objectLocalFilePath, r.SourceBucket, objectKey)
-				if err != nil {
-					return errors.Cause(err)
-				}
+			err = s3.DownloadFile(objectShasumFileLocalFilePath, r.SourceBucket, objectShasumFileKey)
+			if err != nil {
+				return errors.Cause(err)
 			}
 		}
 	}
-	fmt.Printf("%s Successfully downloaded artifacts\n", constants.SuccessIcon)
+
+	return nil
+}
+
+func handleManifestDownload(_ context.Context, r *releasetypes.ReleaseConfig, artifact releasetypes.Artifact, s3Retrier *retrier.Retrier) error {
```

Comment:
Is the context.Context param here being used? I'm wondering if these can be removed.

Reply:
Responded [above](https://github.com/aws/eks-anywhere/pull/7186#discussion_r1433300878).